### PR TITLE
Adding Auth sample

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: auth/Program.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    public class AuthSample
+    {
+        private const string usage = @"Usage:AuthSample <bucket_name>";
+        // [START build_service]
+        /// <summary>
+        /// Creates an authorized Cloud Storage client service using Application 
+        /// Default Credentials.
+        /// </summary>
+        /// <returns>an authorized Cloud Storage client.</returns>
+        public static async Task<StorageService> CreateAuthorizedClientAsync()
```
Comment (SurferJeffAtGoogle) [inline auth/Program.cs]: The new policies we adopted say no async methods in csharp-docs samples.
Also, drop private and static where ever possible.

**Comment by SurferJeffAtGoogle** [comment]
Update buildAndRunTests.bat.

File: auth/Program.cs (hunk 1 above)
Comment (remi) [inline auth/Program.cs]: `CodeFormatter.exe` adds explicit visibility keywords.  Is there a `/rule:` option to *not* do this?

File: auth/Program.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        }
+        // [END build_service]
+
+        // [START list_storage_bucket_contents]
+        /// <summary>
+        /// List the contents of a Cloud Storage bucket.
+        /// </summary>
+        /// <param name="bucket">the name of the Cloud Storage bucket.</param>
+        ///<returns>a list of the contents of the specified bucket.</returns>
+        public Objects ExecuteApiRequest(
```
Comment (SurferJeffAtGoogle) [inline auth/Program.cs]: Can we rename this to something more specific like ListBuckets()?  If you're trying to communicate that this function is the one that actually calls our API, then maybe name it CallListBuckets()?

**Comment by SurferJeffAtGoogle** [comment]
LGTM.

**Pull request merged**